### PR TITLE
Clean up sources docs

### LINKED ISSUE #4303: Cleanup PR #4301
Edit #4301 for style and fix formatting. Ensure the text follows the Knative Style Guide.

### PR DESCRIPTION
**Pull request by snneji**
Fixes formatting in ApiServerSource docs and includes tweaks to other sources docs to make them more consistent.

Fixes #4303

### CODE REVIEW

File: docs/developer/eventing/sources/containersource/reference.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # ContainerSource reference
 
+![API version v1](https://img.shields.io/badge/API_Version-v1-red?style=flat-square)
```
Comment (lionelvillard) [inline docs/developer/eventing/sources/containersource/reference.md:3]: the badge should be green (v1 API). `API_Version-v1-red` -> `API_Version-v1-green`

**Comment by lionelvillard** [comment]
To be consistent, both `Creating a PingSource object` and `Creating a ContainerSource object` sections should probably be sub-sections in their respective section.

**Comment by lionelvillard** [comment]
@snneji can you also search from `gcr.io/knative-releases/knative.dev/eventing-contrib/cmd/event_display` and replace all instances to `gcr.io/knative-releases/knative.dev/eventing/cmd/event_display`? I realize this is applicate to the entire doc.

File: docs/developer/eventing/sources/containersource/reference.md (hunk 1 above)
Comment (snneji) [reply docs/developer/eventing/sources/containersource/reference.md:3]: Thanks. I'll update the remaining red badges.

**Comment by snneji** [comment]
> To be consistent, both `Creating a PingSource object` and `Creating a ContainerSource object` sections should probably be sub-sections in their respective section.

They used to be in their own section in the left navigation but I think they have implemented a new version of MkDocs that changes the way the site deals with README pages. Probably what needs to be done now is to create proper overview pages for PingSource and ContainerSource so the "Creating..." pages can be inside the section again. 

I'll make a new issue for this here: #4305

**Pull request merged**